iceoryx_posh/source/mepoo/shm_safe_unmanaged_chunk.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/internal/mepoo/shm_safe_unmanaged_chunk.hpp"
+
+
+namespace iox
+{
+namespace mepoo
+{
+static_assert(sizeof(ShmSafeUnmanagedChunk) <= 8U, "The ShmSafeUnmanagedChunk size must not exceed 64 bit!");
```

Comment:
Why that? Is it caused by SoFi which then would not be lock-free anymore? Maybe we should add the reasoning here as a comment.

Reply:
to prevent Frankenstein Objects when objects are copied, like in the SoFi where we use memcpy. I can add a note.

Reply:
Ah but this is not a problem with memcpy! Yeah you would have a Frankenstein object but since you are using only memcpy and never call any methods on the object itself it is alright.